packages/indy-vdr/src/dids/IndyVdrIndyDidRegistrar.ts

```diff
@@ -62,11 +63,12 @@ export class IndyVdrIndyDidRegistrar implements DidRegistrar {
     }
 
     try {
-      const { namespace, id: submitterId } = parseIndyDid(submitterDid)
+      // Parse submitterDid and extract namespace based on the submitter did
+      const { namespace: submitterNamespace, id: submitterDidIdentifier } = parseIndyDid(submitterDid)
+      const submitterSigningKey = await verificationKeyForIndyDid(agentContext, submitterDid)
```

Comment:
Nice! I didn't want to call so many APIs from this class but it is certainly more clear for this Registrar API